### PR TITLE
Content management hibernate entities

### PR DESCRIPTION
**Pull request by hustodemon**
## What does this PR change?

Adds hibernate entities and boilerplate code for Content Management feature.

The methods can change based on the future business logic - for now it's mostly CRUD.

## GUI diff

No diff.

- [x] **DONE**

## Documentation
- No documentation needed: only backend code, no user functionality in this PR.

- [x] **DONE**

## Test coverage
- Unit tests were added

- [x] **DONE**

## Links

Tracks https://github.com/SUSE/spacewalk/issues/6505

- [x] **DONE**

## Changelogs

Copy the following sentence as a new comment if you don't need changelog entries:

> gitarro no changelog needed !!!

If the test `changelog_test`already ran, then add another new comment with the following text:

> gitarro rerun changelog_test !!!

### CODE REVIEW

**Comment by hustodemon** [comment]
gitarro no changelog needed !!!

**Comment by mcalmer** [comment]
@hustodemon - some additional ideas I got:

- AFAIK we talked about a "status" for an Environement. Building and promoting may take some time so it makes sense to add a DB field for it. But this can be added later when you implement "build" and "promote".
- When you create an EnvironmentTarget you should "create" the Channel internal to stick to the naming rules. Adding an arbitrary channel is not what we want.
- Project <=> Filter is defined as ManyToMany but only on one side (Project). In best case we do not need the bi-directional way. Only enhance it if you really need it.

File: java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProject.java [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+        historyEntries = historyEntriesIn;
+    }
+
+    /**
+     * Add a history entry
+     *
+     * @param entry - the history entry
+     */
+    public void addHistoryEntry(ContentProjectHistoryEntry entry) {
+    }
```
[the remark this answers is not in the document]
Comment (hustodemon) [reply java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProject.java]: Implemented in the `ContentProjectFactory`. I forgot to remove it from here.

File: java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProjectFactory.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+     *
+     * @param project - Content Project
+     * @param entry - the history entry
+     */
+    public static void addHistoryEntryToProject(ContentProject project, ContentProjectHistoryEntry entry) {
+        List<ContentProjectHistoryEntry> entries = project.getHistoryEntries();
+        entry.setVersion(entries.size() == 0 ? 1 : entries.get(entries.size() - 1).getVersion() + 1);
+        entry.setContentProject(project);
+        save(entry);
+        entries.add(entry);
```
[the remark this answers is not in the document]
Comment (hustodemon) [reply java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProjectFactory.java]: Right. If we save an entry, let's also save the project.

File: java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProjectHistoryEntry.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        created = createdIn;
+    }
+
+    /**
+     * Gets the user.
+     *
+     * @return user
+     */
+    @ManyToOne(targetEntity = UserImpl.class)
+    public User getUser() {
```
[the remark this answers is not in the document]
Comment (hustodemon) [reply java/code/src/com/redhat/rhn/domain/contentmgmt/ContentProjectHistoryEntry.java]: It points to the user via their `id`. If the user gets deleted -> it gets set to null. I covered the behavior in tests ([testProjectHistory](https://github.com/uyuni-project/uyuni/pull/413/files#diff-8d24f06847f64fe83a2d7b69855f8d85R349)).

**Comment by hustodemon** [comment]
Thanks for the review @mcalmer . I addressed the inline comments. I'll move the "general" comments to corresponding issue(s).

**Pull request merged**